openassessment/fileupload/api.py

```diff
@@ -51,14 +51,19 @@ def get_student_file_key(student_item_dict, index=0):
     return key_template.format(index=index, **student_item_dict)
 
 
-def _safe_load_json_list(field):
+def _safe_load_json_list(field, log_error=False):
     """
     Tries to load JSON-ified string,
     returns an empty list if we try to load some non-JSON-encoded string.
     """
     try:
         return json.loads(field)
     except ValueError:
+        if log_error:
```

Comment:
I guess it should help if we log the exception every time, so I don't think the param passed here is useful. Please log the exception here.

Reply:
@awaisdar001 This method also acts as a backward-compatible fix. The file API was recently added with some additional file info(filename & sizes) that don't exist in submissions prior to that. Since data is non-existent, the `ValueError` raised handles it by returning an empty list and not interrupting regular workflow. Logging every time will flood the logs, most of whom will be unnecessary. This is precisely the reason I went with logging during the user state workaround, to give an exact piece of information, when activated.